pyidi/methods/_simplified_optical_flow.py

```diff
@@ -172,15 +172,20 @@ def pixel_shift_fun(self, i, points, image_shape):
             warnings.warn('Displacement is going outside of the image range! The valid points are saved in self.method.valid_points')
         self.displacements[~self.valid_points, i, :] = np.nan
 
-    def reference(self, images, subset_size):
+    def _reference(self):
         """Calculation of the reference image, image gradients and gradient amplitudes.
 
         :param images: Images to average. Usually the first 100 images.
         :param subset_size: Size of the subset to average.
         :return: Reference image, image gradient in 0 direction, image gradient in 1 direction, gradient magnitude
         """
-        reference_image = np.mean([self.subset(image_, subset_size)
-                                   for image_ in images], axis=0)
+        if self.pixel_shift:
+            reference_image = self.subset(self.video.reader.get_frame(self.reference_range[0]), self.subset_size)
+        else:
+            reference_image = np.zeros((self.video.reader.image_height, self.video.reader.image_width), dtype=float)
+            for frame in range(self.reference_range[0], self.reference_range[1]):
+                reference_image += self.subset(self.video.reader.get_frame(frame), self.subset_size)
+            reference_image /= (self.reference_range[1] - self.reference_range[0])
 
```

Comment:
@itomac I can't make any changes, but something like this should do it:
```
for frame in range(self.reference_range[0], self.reference_range[1]):
    reference_image += self.video.reader.get_frame(frame)
reference_image /= (self.reference_range[1] - self.reference_range[0])
reference_image = self.subset(reference_image, self.subset_size)
```

Reply:
Great, thanks for update.